fdb-extensions/src/main/java/com/apple/foundationdb/async/hnsw/AccessInfo.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * AccessInfo.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2025 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.async.hnsw;
+
+import com.apple.foundationdb.linear.RealVector;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+class AccessInfo {
+    @Nonnull
+    private final EntryNodeReference entryNodeReference;
+
+    private final long rotatorSeed;
+
+    @Nullable
+    private final RealVector centroid;
```

Comment:
Maybe this will become clearer later, but my understanding was that we may want to modify the centroid over time in response to the contents of the HNSW. Will we need to rework this so that we don't have one that is baked into the access info for the layer?